types/helper.go

```diff
@@ -23,31 +23,40 @@ import (
 	"github.com/pingcap/errors"
 )
 
-// RoundFloat rounds float val to the nearest even integer value with float64 format, like MySQL Round function.
-// RoundFloat uses default rounding mode, see https://dev.mysql.com/doc/refman/5.7/en/precision-math-rounding.html
-// so rounding use "round to nearest even".
-// e.g, 1.5 -> 2, -1.5 -> -2.
-func RoundFloat(f float64) float64 {
-	return math.RoundToEven(f)
-}
-
-// Round rounds the argument f to dec decimal places.
+// RoundFloat rounds the argument f to dec decimal places using "round to nearest even" rule.
 // dec defaults to 0 if not specified. dec can be negative
-// to cause dec digits left of the decimal point of the
-// value f to become zero.
-func Round(f float64, dec int) float64 {
+// see https://dev.mysql.com/doc/refman/5.7/en/precision-math-rounding.html
+func RoundFloat(f float64, dec int) float64 {
 	shift := math.Pow10(dec)
 	tmp := f * shift
 	if math.IsInf(tmp, 0) {
 		return f
 	}
-	result := RoundFloat(tmp) / shift
+	result := math.RoundToEven(tmp) / shift
 	if math.IsNaN(result) {
 		return 0
 	}
 	return result
 }
 
+// RoundInt rounds the argument i to dec decimal places using "round half up" rule.
+// dec defaults to 0 if not specified. dec can be negative
+func RoundInt(i int64, dec int) int64 {
```

Comment:
We should consider unsigned int range and overflow.
```
MySQL [test]> select round(9223372036854775808,-3);
+-------------------------------+
| round(9223372036854775808,-3) |
+-------------------------------+
|           9223372036854776000 |
+-------------------------------+
1 row in set (0.000 sec)

MySQL [test]> 
MySQL [test]> select round(-9223372036854775808,-3);
ERROR 1690 (22003): BIGINT value is out of range in 'round(-(9223372036854775808),-(3))'
```

Reply:
thanks for suggestion, i will check

Reply:
```sql
select round(1234567890123456789012345678901234567890, -30);
# output: 1234567890000000000000000000000000000000
```
i found both `mysql 8` and `tidb v5.1.1`have above result, 
and `1234567890123456789012345678901234567890` is clearly much larger than `math.MaxInt64`,
so it cloud not be the param or return value of `evalInt`,
i believe above case and unsigned int should have processed in some other place,
so i will not process these case in `roundInt`

Reply:
I don't think so. 
select round(1234567890123456789012345678901234567890, -30); This case is decimal round, not int.

We can create a table with bigint field and insert `-9223372036854775808`, then round it should throw the error.

Reply:
Maybe we can try to port mysql's logic https://github.com/mysql/mysql-server/blob/beb865a960b9a8a16cf999c323e46c5b0c67f21f/sql/item_func.cc#L3413